src/Steppenwolf.Services/Data/BlogPostController.cs

```diff
@@ -12,52 +12,93 @@ namespace Steppenwolf.Services.Data
 {
     public class BlogPostController
     {
-        private readonly IRepository<BlogPostEntity> repository;
+        private readonly IRepository<BlogPostEntity> blogPostRepository;
+        private readonly ICategoryRepository categoryRepository;
         private readonly IMapper mapper;
 
-        public BlogPostController(IRepository<BlogPostEntity> repository, IMapper mapper)
+        public BlogPostController(
+            IRepository<BlogPostEntity> blogPostRepository, 
+            ICategoryRepository categoryRepository,
+            IMapper mapper)
         {
-            this.repository = repository;
+            this.blogPostRepository = blogPostRepository;
+            this.categoryRepository = categoryRepository;
             this.mapper = mapper;
         }
 
         public Task<BlogPost> GetById(Guid id)
         {
-            var blog = this.repository.Query()
+            var blog = this.blogPostRepository.Query()
                 .Include(r => r.Author)
+                .Include(r => r.BlogCategoryEntities)
+                .ThenInclude(r => r.Category)
                 .FirstOrDefault(b => b.Id == id);
             return Task.FromResult(this.mapper.Map<BlogPost>(blog));
         }
         
         // Todo take user id from bearer when migrated to web assembly
         public async Task<Guid> Upsert(BlogPost blogPost, string userId)
         {
+            Guid postId;
+            
             var blog = this.mapper.Map<BlogPostEntity>(blogPost);
-            var exist = blog.Id != default && this.repository.Query(true).Any(b => b.Id == blog.Id);
-            if (exist)
+            var existingPost = blog.Id != default 
+                ? await this.blogPostRepository
+                    .Query(false)
```

Comment:
Does it work for update without tracking?

Reply:
Yes, I tested both creation and deletion